Sources/Peripheral/CharacteristicValueUpdateEventData.swift

```diff
@@ -9,9 +9,28 @@ public struct CharacteristicValueUpdateEventData {
     /// to avoid data races, given that the underlying CBCharacteristic's value might change
     /// before clients receive the publisher's value.
     public let value: Data?
-    
-    init(characteristic: Characteristic) {
+    /// The error that occurred during the value update in `peripheral(_:didUpdateValueFor:error:)`
+    let error: Error?
+
+    /**
+     Returns the updated value of the characteristic, or throws if there was an error.
+     This value is stored separately from the characteristic itself to avoid data races,
+     since the underlying CBCharacteristic's value may change before clients have a chance
+     to receive the published value.
+     - Throws: The error that occurred during the value update in `peripheral(_:didUpdateValueFor:error:)`, if any.
+     - Returns: The value of the characteristic at the time of the update, or nil if unavailable.
+     */
+    public func getValue() throws -> Data? {
```

Comment:
I think it would be slightly confusing to have `value` and `getValue`, do you mind renaming to something like `valueOrError`?